add refresh token support and use in example

noonhub opened the pull request.

slobak commented on src/Asana/Dispatcher/OAuthDispatcher.php:
```diff
@@ -2,7 +2,7 @@
 namespace Asana\Dispatcher;
 
-use \OAuth2;
+use OAuth2;
```
I thought there was a reason for the leading backslash, like it is explicitly starting from the root namespace. Do we not need this?

noonhub replied:
The manual claims they are unnecessary. I was not part of any previous discussion where it was determined they were. It seems to work without it.

slobak commented on src/Asana/Dispatcher/OAuthDispatcher.php:
```diff
@@ -39,10 +40,23 @@ public function fetchToken($code)
         return $this->accessToken;
     }
 
+    public function refreshToken()
```
maybe we should call this `refreshAccessToken()` since we're using "refresh" as a verb? Based on its name it sounds like it might be "getting the refresh token".

Also, should we add comments to this public method?

slobak commented on examples/example-oauth.php:
```diff
@@ -44,13 +42,29 @@
+// access tokens will expire, use a refresh token to get a fresh access token
+$token = $client->dispatcher->refreshToken();
+
+echo "Your new access token is: " . json_encode($token) . "\n";
```
Yeah, this is friendlier. :)

noonhub commented:
I made a few changes
* Leading slash, it seems to work w/o it and the manual indicates they are unnecessary. Unless you feel otherwise I think we're okay.
* Changed the method name but also added automatic refresh support.
* Adding comments seems to be a library wide issue so: https://github.com/Asana/php-asana/issues/17
* I would also like to fix https://github.com/Asana/php-asana/issues/18
